src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -474,9 +474,13 @@ void DiscoveryImplPlatform::HandleNodeResolve(void * context, MdnsService * resu
 
     if (result->mAddress.HasValue())
     {
-        data.ipAddress[data.numIPs++] = result->mAddress.Value();
+        data.ipAddress[data.numIPs]   = result->mAddress.Value();
+        data.interfaceId[data.numIPs] = result->mInterface;
+        data.numIPs++;
```

Comment:
Could this overrun the array? There are no bound checks on `data.numIPs` versus `ArraySize(data.ipAddress)`, etc